File: REST_API/Models/UserModel.js
const mongoose = require('mongoose');

const UserRoleEnum = {
    Admin: 0,
    Faculty: 1,
    Student: 2
};

const userSchema = new mongoose.Schema({
    UserId: String,
    FirstName: String,
    LastName: String,
    Password: String,
    PasswordSalt: String,
    PersonalEmail: String,
    OfficialEmail: String,
    UserName: String,
    MobileNumber: String,
    UserRoleEnum: {
        type: Number,
        enum: Object.values(UserRoleEnum), 
        default: UserRoleEnum.Admin
    },
    UserRoleEnumDisplayname: String,
    RememberMe: Boolean
});

module.exports = mongoose.model('UserModel', userSchema);

File: WebApp/Models/UserModels/UserModel.cs
using System.ComponentModel;

namespace WebApp.Models.UserModels
{
    public class UserModel
    {
        [DisplayName("User")]
        public Guid UserId { get; set; }

        [DisplayName("First Name")]
        public string FirstName { get; set; }

        [DisplayName("Last Name")]
        public string LastName { get; set; }

        [DisplayName("Password")]
        public string Password { get; set; }

        public string PasswordSalt { get; set; }

        [DisplayName("Personal Email")]
        public string PersonalEmail { get; set; }

        [DisplayName("Officeal Email")]
        public string OfficialEmail { get; set; }

        [DisplayName("User Name")]
        public string UserName { get; set; }

        [DisplayName("Mobile Number")]
        public string MobileNumber { get; set; }

        [DisplayName("User Role Enum")]
        [DefaultValue(UserRoleEnum.Admin)]
        public UserRoleEnum UserRoleEnum { get; set; }

        [DisplayName("User Role")]
        public string UserRoleEnumDisplayname
        {
            get { return Enum.GetName(typeof(UserRoleEnum), this.UserRoleEnum); }
        }

        [DisplayName("Remember Me")]
        public bool RememberMe { get; set; }

    }
}
